src/nvim/search.c

```diff
@@ -2411,6 +2412,18 @@ static int cls(void)
 {
   int c;
 
+  // Treat concealed characters like whitespace.
+  if (curwin->w_p_cole > 0 && curwin->w_cursor.col > 0
+      && curbuf->b_p_smc > 0 && curwin->w_cursor.col < curbuf->b_p_smc
+      && conceal_cursor_line(curwin)) {
+    syn_get_id(curwin, curwin->w_cursor.lnum, curwin->w_cursor.col,
+               false, NULL, false);
```

Comment:
No, I actually don't see what the call to `syn_get_id` is accomplishing. I dug in a bit more and see that (surprise) `sys_get_id` has more side effects than I realized, but `syntax_start()` is documented as happening once per line already, so again I wonder why the need to call `syn_get_id` here?